### PR TITLE
Find data filter functionality #4jbra0

### PR DESCRIPTION
**Pull request by cameronbaney**
# Description

The purpose of this PR is to add filter functionality to the Find Data section. This is for content from both contentful and the Blackfynn API

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

## Datasets (Blackfynn API)
- Go to the Find Data page
- On the Datasets tab, click on the "Filters" button
- Select "Stomach" and hit apply
- The search should update and show results for datasets tagged "Stomach"
- Click on the "Filters" button. Select "Tag1" and hit apply
- The search should update and show no results
- Using the tags above the table, click on the X next to "Tag1"
- The search should update and show results for datasets tagged "Stomach"
- Click on the "Filters" button. Click the "Clear All" button
- The search should update and show you all results for Datasets

## Organs (Contentful API)
- Go to the Find Data Page
- On the Organs tab, click on the "Filters" button
- Select "Tag1"
- The search should update and show results for datasets tagged "Tag1"
- Click the "Filters" button
- Select "Heart" 
- The search should update and show no results
- Using the tags above the table, click on the X next to "Heart"
- The search should update and show results for datasets tagged "Tag1"

## Simulations (Blackfynn API)
- Go to the Find Data page
- On the Simulations tab, click on the "Filters" button
- Select "Stomach" and hit apply
- The search should update and show no results

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: pages/data/index.vue [hunk 1]
```diff
@@ -241,6 +247,11 @@ export default {
         url += `&query=${query}`
       }
 
+      const tags = pathOr('', ['query', 'tags'], this.$route)
```
Comment (matthewtgilbride) [inline pages/data/index.vue]: nit:

`const tags = this.$route.query.tags || ''`

File: pages/data/index.vue [hunk 2]
```diff
@@ -353,10 +371,34 @@ export default {
       this.$router.replace({ query: { type: firstTabType } })
     } else {
       this.fetchResults()
+      // this.fetchFilters()
```
Comment (matthewtgilbride) [inline pages/data/index.vue]: remove comment?

File: pages/data/index.vue [hunk 3]
```diff
@@ -353,10 +371,34 @@ export default {
...
+     * Set active filters based on the query params
+     * @params {Array} filters
+     * @returns {Array}
+     */
+    setActiveFilters: function(filters) {
+      const tags = compose(
+        split(','),
+        toLower,
+        pathOr('', ['query', 'tags'])
+      )(this.$route)
```
Comment (matthewtgilbride) [inline pages/data/index.vue]: `(this.$route.query.tags || '').toLowerCase().split(',')`

Comment (matthewtgilbride) [reply pages/data/index.vue]: I am pretty sure `query` is always defined.  When there is no query it is an empty object.  Correct me if I'm wrong.

File: pages/data/index.vue [hunk 4]
```diff
@@ -479,6 +526,28 @@ export default {
...
+      })
+
+      const queryParamTags = pathOr('', ['query', 'tags'], this.$route)
+      if (equals(filterVals, queryParamTags.split(','))) {
+        return
+      }
+
+      const tags = { tags: filterVals.join(',') }
+
+      const query = mergeLeft(tags, this.$route.query)
```
Comment (matthewtgilbride) [inline pages/data/index.vue]: `const query = { ...this.$route.query, ...tags }`

Comment (matthewtgilbride) [reply pages/data/index.vue]: I'm pretty sure `mergeLeft` behaves the same way as the spread operator after reading the doc again.  I could be wrong...

**Comment by cameronbaney** [comment]
@matthewtgilbride I addressed your feedback.

File: pages/data/index.vue (hunk 3 above)
Comment (cameronbaney) [reply pages/data/index.vue]: You are correct, and this is nice: `(this.$route.query.tags || '')`

**Pull request merged**